core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -904,7 +793,9 @@ public Transaction replaceTransaction() {
       LoadTableResponse response = loadInternal(context, ident, snapshotMode);
       String fullName = fullTableName(ident);
 
-      AuthSession session = tableSession(response.config(), session(context));
+      Map<String, String> tableConf = response.config();
+      AuthSession parent = authManager.contextualSession(context, catalogAuth);
```

Comment:
nit: we're calling this `contextualSession` everywhere else, so would be good to do the same here